docs/source/swath.rst

```diff
@@ -268,10 +268,74 @@ Click images to see the full resolution versions.
 
 The *perceived* sharpness of the bottom image is lower, but there is more detail present.
 
+
+XArrayResamplerBilinear
+***********************
+
+**bilinear.XArrayResamplerBilinear** is a class that handles bilinear interpolation for data in
+`xarray.DataArray` arrays.  The parallelisation is done automatically using `dask`.
+
+.. doctest::
+
+>>> import dask.array as da
+>>> from xarray import DataArray
+>>> from pyresample.bilinear.xarr import XArrayResamplerBilinear
+>>> from pyresample import geometry
+>>> target_def = geometry.AreaDefinition('areaD',
+...                                      'Europe (3km, HRV, VTC)',
+...                                      'areaD',
+...                                      {'a': '6378144.0', 'b': '6356759.0',
+...                                       'lat_0': '50.00', 'lat_ts': '50.00',
+...                                       'lon_0': '8.00', 'proj': 'stere'},
+...                                      800, 800,
+...                                      [-1370912.72, -909968.64,
+...                                       1029087.28, 1490031.36])
+>>> data = DataArray(da.from_array(np.fromfunction(lambda y, x: y*x, (500, 100))), dims=('y', 'x'))
+>>> lons = da.from_array(np.fromfunction(lambda y, x: 3 + x * 0.1, (500, 100)))
+>>> lats = da.from_array(np.fromfunction(lambda y, x: 75 - y * 0.1, (500, 100)))
+>>> source_def = geometry.SwathDefinition(lons=lons, lats=lats)
+>>> resampler = XArrayResamplerBilinear(source_def, target_def, 30e3)
+>>> resampler.get_bil_info()
+>>> result = resampler.get_sample_from_bil_info(data)
```

Comment:
The separate `get_bil_info()` step is necessary when caching the resampling info. In the follow-ups I move the caching from Satpy to Pyresample, so with that the caching step can be shown here. In another follow-up I'll add `resampler.resample()` method that wraps `resampler.get_bil_info()` and `resampler.get_sample_from_bil_info()` together. It could also have the `cache_dir` kwarg so that there'd be no need to call it separately.